### PR TITLE
Update the syntax docs for `REQUIRED`/`OPTIONAL`.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Update the DDL and SDL docs for links and properties.

### CODE REVIEW

File: docs/edgeql/ddl/links.rst [hunk 1]
```diff
@@ -80,6 +82,11 @@ Parameters
     the *required* attribute, i.e it is not possible to make a
     required link non-required by extending it.
 
+:eql:synopsis:`OPTIONAL`
+    If specified, the link is considered *optional* for the parent
```
Comment (1st1) [inline docs/edgeql/ddl/links.rst:86]: OPTIONAL is default, so it doesn't matter if it's specified or not. Please reflect that it's the default.

File: docs/edgeql/ddl/props.rst [hunk 2]
```diff
@@ -78,6 +79,11 @@ Parameters
     inherit the *required* attribute, i.e it is not possible to make a
     required property non-required by extending it.
 
+:eql:synopsis:`OPTIONAL`
+    If specified, the property is considered *optional* for the parent
+    object type, i.e. it is possible for the property to resolve to an
+    empty value.
```
Comment (1st1) [inline docs/edgeql/ddl/props.rst:85]: ditto

File: docs/edgeql/ddl/links.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/edgeql/ddl/links.rst]: Updated to clarify it's the default.

File: docs/edgeql/ddl/props.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/edgeql/ddl/props.rst]: OK

**Pull request merged**